test/recipes/20-test_cli_fips.t

```diff
@@ -339,6 +357,32 @@ SKIP : {
                      '-out', $testtext_prefix.'.fail.priv.pem'])),
            $testtext);
 
+        $testtext = $testtext_prefix.': '.
+            'Generate a key with non-FIPS parameters using non-FIPS property'.
+            ' query (dsaparam)';
+        ok(run(app(['openssl', 'dsaparam', '-provider', 'default',
+                    '-propquery', '?fips!=yes',
```

Comment:
I'm not sure this is what you want to do (i.e. the `?` preceding the query string. It means that it is "optional but preferred" (see https://www.openssl.org/docs/man3.0/man7/property.html). Which means it could *still* use the FIPS module.

Reply:
I'm aware of that, and it was a deliberate choice.

The issue with using `fips!=yes` is that the store loaders are tagged `fips=yes` regardless of which provider they're in, i.e., also in the default provider. See

 - https://github.com/openssl/openssl/blob/openssl-3.0/providers/defltprov.c#L474-L481
 - https://github.com/openssl/openssl/blob/openssl-3.0/providers/baseprov.c#L85-L92

Using `fips!=yes` then leads to the following error message for some of those:
```
Could not open file or uri for loading private key from root.key
00DC3E2F857F0000:error:16000069:STORE routines:ossl_store_get0_loader_int:unregistered scheme:crypto/store/store_register.c:237:scheme=file
00DC3E2F857F0000:error:1608010C:STORE routines:inner_loader_fetch:unsupported:crypto/store/store_meth.c:357:No store loader found. For standard store loaders you need at least one of the default or base providers available. Did you forget to load them? Info: Global default library context, Scheme (file : 0), Properties (fips!=yes)
```

Should we modify the default provider to define those without `fips=yes` to avoid this?